### PR TITLE
Arrays

### PR DESCRIPTION
**Pull request by jamescheney**
This PR is work in progress to merge concurrent progress.  Will comment again when ready for review.

### CODE REVIEW

**Comment by jstolarek** [comment]
James, if you have a fix for build errors could you push it? I'd like to start working on reducing computational complexity of slicing and I want to start from `arrays` branch (even if the arrays don't work yet).

**Comment by jamescheney** [comment]
All tests passing, added some array tests, now ready for review and merge (or just to be used to write more interesting examples)

File: examples/array.tml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+let t = trace (
+let x = array(1,1000) in
```
Comment (jstolarek) [inline examples/array.tml]: What are the arguments to array?

File: lib/Language/Slicer/Resugar.hs [hunk 2]
```diff
@@ -74,11 +77,16 @@ instance Uneval Trace Exp where
     uneval (TIfElse t t2)    = EIf (uneval t) bot (uneval t2)
     uneval (TIfExn t)        = EIf (uneval t) bot bot
     uneval (TOp _ f es)      = EOp f (map uneval es)
-    uneval (TCall t1 t2 _ _) = EApp (uneval t1) (uneval t2)
+    uneval (TCall t1 t2 _ _) = EApp (uneval t1) (uneval t2)  -- jrc: ???
```
Comment (jstolarek) [inline lib/Language/Slicer/Resugar.hs]: ?

**Comment by jamescheney** [comment]
Sorry about the errors.  Everything should build/test correctly now and I am done making changes (unless anything is broken).

File: examples/array.tml (hunk 1 above)
Comment (jamescheney) [reply examples/array.tml]: array(n,v) creates an array of n elements with initial value v.

File: tests/SlicerTestSuite.hs [hunk 3]
```diff
@@ -11,11 +11,15 @@ tests :: TestTree
 tests =
     testGroup "Run TML files" (map runTMLTest
       [ "abs"
+      , "array"
```
Comment (jstolarek) [inline tests/SlicerTestSuite.hs:14]: One thing to remember is that new examples should be added to a list of tests AND to a list of benchmarks in `bench` directory. I will fix these now, so no worries - just something to remember in the future :-)  I know this duplication is irritating but I don't see a way of avoiding it.

**Comment by jstolarek** [comment]
Rebased and merged.

**Pull request closed without merging**